fontforge/parsettf.c

```diff
@@ -3320,6 +3333,10 @@ return( NULL );
 	offsets[i] = getoffset(ttf,offsize);
     dicts = malloc((count+1)*sizeof(struct topdicts *));
     for ( i=0; i<count; ++i ) {
+	if (fontnames != NULL && fontnames[i] == NULL) {
+		LogError(_("Number of CFF font names is less than dict size: %d < %d"), i, count);
+		break;
+	}
```

Comment:
Not quite; `dicts` is NULL terminated, which happens at the conclusion of the loop (line 3347).